presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -949,6 +951,7 @@ public boolean isImpersonationEnabled()
             cacheBuilder = cacheBuilder.refreshAfterWrite(refreshMillis.getAsLong(), MILLISECONDS);
         }
         cacheBuilder = cacheBuilder.maximumSize(maximumSize);
+        cacheBuilder.recordStats();
```

Comment:
```suggestion
        cacheBuilder = cacheBuilder.maximumSize(maximumSize)
            .recordStats();
```

Reply:
Does it have influence on performance?

Reply:
>Does it have influence on performance?

CachingHiveMetastore used during query planning to get information about tables/partitions/etc. `recordStats()` pattern is used in `ExpressionCompiler`, `JoinCompiler`, other `*Compilter`classes which called much more often  then CachingHiveMetastore (there more expressions that tables in each query). As a result I don't expect material performance impact. 

Also, number of events should be close to number of events produced by `ThriftHiveMetastore` when no caching is used that already exposed via JMX.

Still trying to figure out how to run benchmarking and coverage for these class.

Reply:
> Still trying to figure out how to run benchmarking and coverage for these class.

I don't think we need automated tests for that. It is ok.